### PR TITLE
Add VisionOS support

### PR DESCRIPTION
**Pull request by AnthonyMDev**
This PR adds support or VisionOS by:

- Adding the visionOS platform to the `Package.swift` file
- Adding a visionOS target, test target, test plan, and scheme to the Xcode Project

**When running the tests on visionOS, there is a single failing test. I'm not familiar enough with the library to know what could be causing that failure. If someone is able to point me in the right direction, I'd be happy to get the test passing so this can be merged in.**

### CODE REVIEW

**Comment by AnthonyMDev** [comment]
It looks like this test failure is unrelated to this PR, as [it was also failing in this other PR](https://github.com/stephencelis/SQLite.swift/pull/1224#issuecomment-1764390832).

**Comment by AnthonyMDev** [comment]
@nathanfallet, any chance we could get some eyes on this? We've got users of our SDK asking for VisionOS support, and we are waiting on support in this library to make that work.

**Review by nathanfallet — APPROVED** [review]
Just waiting for checks to run

**Comment by AnthonyMDev** [comment]
Awesome thank you! Looks like the just that one broken test that is failing still.

**Comment by nathanfallet** [comment]
Yes I need to check why; I don't think it's related to visionOS addition, but I don't want to merge if it's failing 😅

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
+    runs-on: macos-14
     steps:
     - uses: actions/checkout@v2
     - name: "Select Xcode"
       # Currently only works with Xcode 14.2:
       # https://github.com/CocoaPods/CocoaPods/issues/11839
       run: |
         xcode-select -p
-        sudo xcode-select -s /Applications/Xcode_14.2.app/Contents/Developer
+        sudo xcode-select -s /Applications/Xcode_15.0.1.app/Contents/Developer
```
Comment (loganblevins) [inline .github/workflows/build.yml:16]: Would suggest 15.2

**Comment by nathanfallet** [comment]
Last thing to fix:
```
-> SQLite.swift/standalone (0.14.1)
    - ERROR | [SQLite.swift/standalone] xcodebuild: Returned an unsuccessful exit code. You can use `--verbose` for more information.
    - NOTE  | [SQLite.swift/standalone] xcodebuild:  note: Using codesigning identity override: -
    - NOTE  | [SQLite.swift/standalone] xcodebuild:  note: Building targets in dependency order
    - NOTE  | [iOS] [SQLite.swift/standalone] xcodebuild:  note: Target dependency graph (4 targets)
    - NOTE  | [SQLite.swift/standalone] xcodebuild:  clang: error: SDK does not contain 'libarclite' at the path '/Applications/Xcode_15.0.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/arc/libarclite_iphonesimulator.a'; try increasing the minimum deployment target
    - NOTE  | [SQLite.swift/standalone] xcodebuild:  Pods.xcodeproj: warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 12.0 to 17.0.99. (in target 'sqlite3' from project 'Pods')
    - NOTE  | [iOS] [SQLite.swift/standalone] xcodebuild:  note: Target dependency graph (3 targets)

[!] SQLite.swift did not pass validation, due to 1 error.
```
I had the warning for `IPHONEOS_DEPLOYMENT_TARGET` 3 times (the two other ones were from 11.0, I updated it in the podspec) but I'm unable to get rid of this last one. (it's a warning but I read the missing `libarclite` was linked to this)

**Comment by groue** [comment]
> Last thing to fix

It's probably https://github.com/CocoaPods/CocoaPods/issues/11839. If you manage to fix it, I'll be happy to know your solution :-)

**Comment by calvincestari** [comment]
@nathanfallet - where are we on getting this resolved and merged?

**Comment by nathanfallet** [comment]
@calvincestari I'm trying to find a solution with cocoapods that currently breaks the project. When that will be fixed, we'll be able to merge it.

**Comment by fenhelix** [comment]
@nathanfallet Any luck or insight on the cocoapods breaking the VisionOS support?

**Comment by calvincestari** [comment]
Thanks for getting this merged @nathanfallet, much appreciated!

**Comment by calvincestari** [comment]
@nathanfallet, another question for this: will the podspec be updated to declare support for visionOS too? When I try push our updated podspec I get an error about incompatibility with SQLite.swift and visionOS. I believe it requires another [deployment target](https://github.com/stephencelis/SQLite.swift/blob/master/SQLite.swift.podspec#L29).
>     - ERROR | [visionOS] [Apollo/SQLite] unknown: Encountered an unknown error (The platform of the target `App` (visionOS 1.0) is not compatible with `SQLite.swift (0.15.1)`, which does not support `visionOS`.) during validation.

**Comment by nathanfallet** [comment]
@calvincestari I'll have a look at that, it might have been missing in the PR

**Pull request merged**